packages/block-library/src/navigation-link/index.php

```diff
@@ -338,7 +338,7 @@ function block_core_navigation_link_register_variation( $variation ) {
 		return;
 	}
 
-	$navigation_block_type->variations[] = $variation;
+	$navigation_block_type->variations = array_merge( $navigation_block_type->variations, $variation );
```

Comment:
This will not fixed the issue as https://github.com/WordPress/gutenberg/blob/2bc59bda8f8bdf1d43b34ff2a23ccb807f6c7214/packages/block-library/src/navigation-link/index.php#L360-L363 they check for variation name and remove it.